main.js

```diff
@@ -3292,11 +3298,20 @@ async function startInstance(id, wakeUp) {
     }
 
     // default: if less than 100 MB log warning, less than 50 MB log error, but check config first
-    if (availableMemMB !== undefined && availableMemMB < (typeof config.system.memLimitError === 'number' ? config.system.memLimitError : 50)) {
-        logger.error(`${hostLogPrefix} Your system has only ${availableMemMB} MB RAM left available and an additional adapter process is started. Please check your system, settings and active instances to prevent swapping and Out-Of-Memory situations!`);
-        logger.error(`${hostLogPrefix} In future versions, the adapter might not be started!`);
-    } else if (availableMemMB !== undefined && availableMemMB < (typeof config.system.memLimitWarn === 'number' ? config.system.memLimitWarn : 100)) {
-        logger.warn(`${hostLogPrefix} Your system has only ${availableMemMB} MB RAM left available and an additional adapter process is started. Please check your system, settings and active instances to prevent swapping and Out-Of-Memory situations!`);
+    if (availableMemMB !== undefined && availableMemMB < (typeof config.system.memLimitWarn === 'number' ? config.system.memLimitWarn : 100)) {
+        if (availableMemMB < (typeof config.system.memLimitError === 'number' ? config.system.memLimitError : 50)) {
+            logger.error(`${hostLogPrefix} Your system has only ${availableMemMB} MB RAM left available and an additional adapter process is started. Please check your system, settings and active instances to prevent swapping and Out-Of-Memory situations!`);
+            logger.error(`${hostLogPrefix} In future versions, the adapter might not be started!`);
+        } else {
+            logger.warn(`${hostLogPrefix} Your system has only ${availableMemMB} MB RAM left available and an additional adapter process is started. Please check your system, settings and active instances to prevent swapping and Out-Of-Memory situations!`);
+        }
+
+        // add it to notifications for popup
+        try {
+            await notificationHandler.addMessage('system', 'memIssues', `Your system has only ${availableMemMB} MB RAM left available and an additional adapter process is started. Please check your system, settings and active instances to prevent swapping and Out-Of-Memory situations!`, `system.host.${hostname}`);
```

Comment:
Maybe (just to be sure) add a check that notificationHandler is defined? ;-)

Reply:
hm why its inside try catch

Reply:
ha ... ok... nice ... yes ok ... :-)

Reply:
lol ;-)